tripper/literal.py

```diff
@@ -0,0 +1,193 @@
+"""Literal rdf values."""
+import re
+import warnings
+from datetime import datetime
+from typing import TYPE_CHECKING
+
+from .namespace import OWL, RDF, RDFS, XSD
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing import Any, Tuple, Union
+
+
+class Literal(str):
+    """A literal RDF value."""
+
+    lang: str
+    datatype: "Any"
+
+    datatypes = {
+        datetime: XSD.dateTime,
+        bytes: XSD.hexBinary,
+        bytearray: XSD.hexBinary,
+        bool: XSD.boolean,
+        int: XSD.integer,
+        float: XSD.double,
+        str: XSD.string,
+    }
+
+    def __new__(cls, value, lang=None, datatype=None):
```

Comment:
> Unsure whether the return type here should actually be `"Literal"` instead?

Yes, it should

Reply:
Mypy is just too stupid to for this. It correctly requires that __new__() should return Literal. But then it complains about line 58 where we assign a string to the return value (which is fully ok since Literal is a subclass of str). Unless there is a way to locally shut up mypy, __new__() should be left without annotations.